ai-compile/vscode-extension/src/codeActionProvider.ts

```diff
@@ -1,9 +1,89 @@
 import * as vscode from "vscode";
+import * as path from "path";
 
-export const diagosticCollection = vscode.languages.createDiagnosticCollection(
-  "pythonedu"
-);
+import * as pymacer from "./pymacer";
 
-export function updateDiagnostics(document: vscode.TextDocument) {
-    
-}
\ No newline at end of file
+export class EduCodeActionProvider implements vscode.CodeActionProvider {
+  public static readonly providedCodeActionKinds = [
+    vscode.CodeActionKind.QuickFix,
+  ];
+
+  public diagnosticCollection = vscode.languages.createDiagnosticCollection(
+    "pythonedu"
+  );
+
+  public codeActions: Map<vscode.Diagnostic, vscode.CodeAction> = new Map();
+  public createDiagnostics: boolean = true;
```

Comment:
Use these flags to decide whether you should show any diagnostics (or actions/fixes). For example, you may not want to show any diagnostics initially until the student clicks the Pencil icon.